### PR TITLE
feat: Add useInfiniteQuery support

### PR DESCRIPTION
**Pull request by 7nohe**
#117 
This new feature will generate a function in infiniteQueries.ts when the name specified by the pageParam option exists in the query parameters and the name specified by the nextPageParam option exists in the response.

Input:
```yml
  /paginated-pets:
    get:
      description: |
        Returns paginated pets from the system that the user has access to
      operationId: findPaginatedPets
      parameters:
        - name: page
          in: query
          description: page number
          required: false
          schema:
            type: integer
            format: int32
        - name: tags
          in: query
          description: tags to filter by
          required: false
          style: form
          schema:
            type: array
            items:
              type: string
        - name: limit
          in: query
          description: maximum number of results to return
          required: false
          schema:
            type: integer
            format: int32
      responses:
        '200':
          description: pet response
          content:
            application/json:
              schema:
                type: object
                properties:
                  pets:
                    type: array
                    items:
                      $ref: '#/components/schemas/Pet'
                  nextPage: 
                    type: integer
                    format: int32
                    minimum: 1
```

Output:
```ts
export const useDefaultServiceFindPaginatedPetsInfinite = <
  TData = InfiniteData<Common.DefaultServiceFindPaginatedPetsDefaultResponse>,
  TError = unknown,
  TQueryKey extends Array<unknown> = unknown[]
>(
  {
    limit,
    tags,
  }: {
    limit?: number;
    tags?: string[];
  } = {},
  queryKey?: TQueryKey,
  options?: Omit<UseInfiniteQueryOptions<TData, TError>, "queryKey" | "queryFn">
) =>
  useInfiniteQuery({
    queryKey: Common.UseDefaultServiceFindPaginatedPetsKeyFn(
      { limit, tags },
      queryKey
    ),
    queryFn: ({ pageParam }) =>
      DefaultService.findPaginatedPets({
        limit,
        page: pageParam as number,
        tags,
      }) as TData,
    initialPageParam: 1,
    getNextPageParam: (response) => (response as { nextPage: number }).nextPage,
    ...options,
  });

```

This can be used as follows:

```tsx
export default function PaginatedPets() {
  const { data, fetchNextPage } = useDefaultServiceFindPaginatedPetsInfinite({
    limit: 10,
    tags: [],
  });

  return ...
}
```

### CODE REVIEW

**Comment by 7nohe** [comment]
npm publish

File: src/createUseQuery.mts [hunk 1]
```diff
@@ -238,18 +244,29 @@ export function createQueryHook({
...
   method: MethodDeclaration;
   className: string;
+  pageParam?: string;
+  nextPageParam?: string;
 }) {
   const methodName = getNameFromMethod(method);
   const customHookName = hookNameFromMethod({ method, className });
   const queryKey = createQueryKeyFromMethod({ method, className });
 
+  const isInfiniteQuery =
```
Comment (seriouslag) [inline src/createUseQuery.mts:263]: should we throw an exception if 'queryString' is not set to 'useInfiniteQuery' and either pageParam or nextPageParam is undefined?

**Review by seriouslag — APPROVED** [review]
PR looks great! I would love to see some updates to the readme.

An infinite scroll example using an Intersection observer would be awesome.
It doesn't have to be in the PR :)

**Comment by HJK181** [comment]
What's the state of the PR?

**Pull request merged**